packages/contracts/contract/src/actions/transactions.rs

```diff
@@ -1,28 +1,171 @@
-use near_sdk::{env, near_bindgen, AccountId, Promise};
+use near_sdk::{env, near_bindgen, AccountId, Promise, json_types::U128, is_promise_success};
 use near_bigint::U256;
 use near_plonk_verifier::{Proof, G1Point};
 
+use crate::*;
 use crate::{
-  Contract, ContractExt,
   hashes::{account_hash, serial_hash},
   events::{event_deposit, event_withdrawal},
 };
 
+const CALLBACK_GAS: Gas = Gas(30_000_000_000_000);
+
+#[ext_contract(ext_self)]
+pub trait ExtSelf {
+  fn withdraw_callback(
+    nullifier_hash: U256,
+    recipient: AccountId,
+    relayer: Option<AccountId>,
+    fee: U256,
+    refund: U256,
+  );
+}
+
 #[near_bindgen]
 impl Contract {
   #[payable]
   pub fn deposit(&mut self, secrets_hash: U256) {
+    assert!(
+      self.currency.is_near(),
+      "contract does not accept NEAR token"
```

Comment:
Should't this message be "Contract only accepts NEAR token" - You are asserting that the currency is NEAR. If it is NEAR, it will not panic, if it is not NEAR it will, and then it will tell the user it does not accept NEAR.

Reply:
You're correct, fixing.